pkg/egressgateway/manager.go

```diff
@@ -197,6 +201,10 @@ func NewEgressGatewayManager(p Params) (out struct {
 		return out, fmt.Errorf("egress gateway requires --%s=\"true\" and --%s=\"true\"", option.EnableIPv4Masquerade, option.EnableBPFMasquerade)
 	}
 
+	if dcfg.EnableIPv6 && !dcfg.EnableIPv6Masquerade {
+		return out, fmt.Errorf("egress gateway requires --%s=\"true\" if --%s=\"true\"", option.EnableIPv6Masquerade, option.EnableIPv6Name)
+	}
+
```

Comment:
I could see this additional requirement causing unexpected issues for users on upgrade (they haven't changed anything in their config, but suddenly the agent refuses to start).

Let's take a follow-up item to make this more graceful (caching the lack of sufficient IPv6 support during startup, and later rejecting any CEGP that lists IPv6 DstCIDRs?). Along with some controlplane testing that demonstrates the desired behavior for an IPv4-only cluster with IPv6 DstCIDRs.

Reply:
I actually removed the checks and made a TODO note for future improvements, will create follow-up issues for the remaining TODO items I marked in the code.